scripts/cl_acquisition.py

```diff
@@ -60,120 +66,268 @@ def __init__(self):
+        pos = numpy.empty((rep[1], rep[0], 2), dtype=numpy.float)
+        posy = pos[:,:, 1].swapaxes(0, 1)  # just a view to have Y as last dim
+        posy[:,:] = numpy.linspace(lim_main[1], lim_main[3], rep[1])
+        # fill the X dimension
+        pos[:,:, 0] = numpy.linspace(lim_main[0], lim_main[2], rep[0])
+
+        return pos
+
+    def get_pixel_size(self) -> Tuple[float, float]:
+        """
+        Computes the pixel size (based on the repetition, roi and FoV of the
+          e-beam). The RepetitionStream does provide a .pixelSize VA, which
+          should contain the same value, but that VA is for use by the GUI.
+        return (float, float): pixel size in m.
+        """
+        epxs = self.escan.pixelSize.value
+        rep = self.repetition
+        roi = self.roi
+        eshape = self.escan.shape
+
+        phy_size_x = (roi[2] - roi[0]) * epxs[0] * eshape[0]
+        phy_size_y = (roi[3] - roi[1]) * epxs[1] * eshape[1]
+        pxsx = phy_size_x / rep[0]
+        pxsy = phy_size_y / rep[1]
+        logging.debug("px size guessed = %s x %s", pxsx, pxsy)
+
+        return (pxsx, pxsy)
+
+    def get_center_pxs(self, datatl):
+        """
+        Computes the center and pixel size of the entire data based on the
+        top-left data acquired.
+        datatl (DataArray): first data array acquired
+        return:
+            center (tuple of floats): position in m of the whole data
+            pxs (tuple of floats): pixel size in m of the sub-pixels
+        """
+        # Compute center of area, based on the position of the first point (the
+        # position of the other points can be wrong due to drift correction)
+        center_tl = datatl.metadata[model.MD_POS]
+        dpxs = datatl.metadata[model.MD_PIXEL_SIZE]
+        tl = (center_tl[0] - (dpxs[0] * (datatl.shape[-1] - 1)) / 2,
+              center_tl[1] + (dpxs[1] * (datatl.shape[-2] - 1)) / 2)
+        logging.debug("Computed center of top-left pixel at at %s", tl)
+
+        # Note: we don't rely on the MD_PIXEL_SIZE, because if the e-beam was in
+        # spot mode (res 1x1), the scale is not always correct, which gives an
+        # incorrect metadata.
+        pxs = self.get_pixel_size()
+
+        rep = self.repetition
+        center = (tl[0] + (pxs[0] * (rep[0] - 1)) / 2,
+                  tl[1] - (pxs[1] * (rep[1] - 1)) / 2)
+        logging.debug("Computed data width to be %s x %s, with center at %s",
+                      pxs[0] * rep[0], pxs[1] * rep[1], center)
+
+        return center, pxs
+
+    def assemble_sem_data(self, data: List[DataArray]) -> DataArray:
+        """
+        Assemble SEM data and metadata
+        :param data: sorted with X fast, Y slow. Each DataArray should be of shape (1,1).
+        :return: DataArray with the correct metadata and 5D shape (111YX)
+        """
+        # get metadata, no need to ask directly to the component because the metadata is already embedded in the first dataset
```

Comment:
Comment line is a bit too long.
```suggestion
        # get metadata, no need to ask directly to the component because 
        # the metadata is already embedded in the first dataset
```